.github/copilot-instructions.md

```diff
@@ -0,0 +1,229 @@
+# GitHub Copilot Instructions for YASSG
+
+## Repository Overview
+
+YASSG (Yet Another Static Site Generator) is a PHP-based static site generator that combines the power of Twig templating with Symfony Encore for asset management and uses YAML databases for content organization.
+
+## Project Architecture
+
+### Core Components
+
+- **Static Site Generator**: Generates static HTML sites from Twig templates
+- **YAML Database**: Uses YAML files for content and data organization
+- **Asset Pipeline**: Symfony Encore for JavaScript/CSS bundling
+- **CLI Interface**: Console commands for site generation and initialization
+- **Multi-environment**: Supports different build environments (dev, prod)
+
+### Key Directories
+
+- `src/` - Core library code (PSR-4: `Sigwin\YASSG\`)
+- `bin/yassg` - Main CLI entry point
+- `config/` - Symfony configuration files
+- `resources/init/` - Project initialization templates
+  - `basic/` - Minimal project setup
+  - `demo/` - Full-featured demo site
+  - `gitlab/` - GitLab CI/Pages configuration
+- `tests/` - Test suite including functional tests
+- `web/` - Web assets and frontend code
+
+## Development Environment
+
+### Prerequisites
+
+- PHP 8.3+ with ext-zlib
+- Composer for dependency management
+- Docker (for development tools)
+- Make for build automation
+
+### Setup
+
+```bash
+# Install dependencies
+composer install
+
+# Run tests
+make test
+
+# Run code analysis
+make analyze
+
+# Format code
+make dist
```

Comment:
`make cs` formats code, `make dist` does *everything*, it's basically a local version of what happens on CI.